pkg/codegen/python/gen.go

```diff
@@ -1896,9 +1897,20 @@ func genPackageMetadata(
 	fmt.Fprintf(w, "from subprocess import check_call\n")
 	fmt.Fprintf(w, "\n\n")
 
-	// Create a constant for the version number to replace during build
-	fmt.Fprintf(w, "VERSION = \"0.0.0\"\n")
-	fmt.Fprintf(w, "PLUGIN_VERSION = \"0.0.0\"\n\n")
+	// If `pkg` supplies a version, use that. Otherwise, create a constant for
+	// the version number to replace during build.
+	packageVersion := "0.0.0"
+	if pkg.Version != nil {
+		packageVersion = pythonVersion(pkg.Version)
+	}
+	pluginVersion := "0.0.0"
+	if version.Version != "" {
+		// This happens in test builds
+		pluginVersion = version.Version
+	}
```

Comment:
I don't think that we want to use `version.Version` here: `"github.com/pulumi/pulumi/pkg/v3/version"` is the version of the Pulumi SDK itself, and is unrelated to the schema version. Should this be

```suggestion
	pluginVersion := "0.0.0"
	if pkg.Version != nil {
		pluginVersion = pkg.Version.String()
	}
```

Reply:
Your right. I got the version of the `pulumi-resource-*` confused with the version of `pulumi` itself.